src/meltano/core/venv_service.py

```diff
@@ -703,12 +704,53 @@ async def handle_installation_error(
         Returns:
             The error that occurred during installation with additional context.
         """
+        self.pip_log_path.parent.mkdir(parents=True, exist_ok=True)
+        try:
+            await self._write_error_log(err)
+        except OSError as log_err:
+            logger.debug(
+                "Failed to write installation error to log file %s: %s",
+                self.pip_log_path,
+                log_err,
+            )
+
+        logger.info(
+            "Logged uv pip install output to %s",
+            self.pip_log_path,
+        )
         return AsyncSubprocessError(
             f"Failed to install plugin '{self.name}'.",
             err.process,
             stderr=await err.stderr,
         )
 
+    async def _write_error_log(self, err: AsyncSubprocessError) -> None:
+        """Write error details to the log file."""
+        stderr_content = await err.stderr
+        await asyncio.to_thread(self._write_error_log_sync, err, stderr_content)
```

Comment:
We use `anyio` elsewhere to interact with the filesystem on an async API, is that possibility here? Or is there a reason this approach is preferred?

Reply:
No particular reason, no! Just unfamiliarity with the codebase. Thanks for the heads up! 🙏🏾 Moved to `anyio` in [3f36eb7](https://github.com/meltano/meltano/pull/9400/commits/3f36eb70ae206714e89521d86c57249f3d30d299).